brave/brave-core
# Brave News Feed Migrator

Resolves https://github.com/brave/brave-browser/issues/24550

This adds a PR for migrating sources which used to be available in `sources.json` but have been removed in `v2`. The migration works by downloading the old sources and adding them to direct feeds, where necessary.

In addition, adds a `BraveNewsV2` flag, which uses the new `global` source.json file. This source file isn't live yet, so to test it, set `--brave-today-host=bn.fallaciousreasoning.nz` when running Brave (the `locales` changes aren't in staging yet, so this is my manual sources update).

It also adds rudimentary support for inferring the current Brave News locale to use. This logic doesn't have any complicated fallbacks (yet) and just falls any unsupported locale to `en_US`. We can make this more complicated but I think that can come in a separate PR.

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Launch Brave, with a new profile `--user-data-dir=/tmp/bn-migrate-sources`
2. Enable Brave News
     - The Brave News feed should load
     - There should be a number of categories
     - There should be a number of sources available in the `All Sources` section
4. Subscribe to `Hollywood Life` in the `Entertainment` category
5. Enable the brave://flags/#brave-news-v2 flag
6. Close Brave
7. Start Brave, with the same profile, and the command line flag `--brave-today-source=brave-today-cdn.bravesoftware.com`. This will point Brave at the staging Brave News environment.
    - The Brave News feed should still load
    - There should be a number of categories in the Customise Brave news dialog
    - There should be a number of sources in the `All Sources` category.
    - `Hollywood Life` should be added as a direct feed now.
